src/stack/mod.rs

```diff
@@ -0,0 +1,142 @@
+use no_std_net::SocketAddr;
```

Comment:
What are your thoughts about potentially even splitting this into a `udp.rs` and `tcp.rs` file as well? In general, I like this cleanup change otherwise.

Reply:
I'm in favor of that actually, but didn't want to shake things up too much and risk push-back :)

Reply:
Let's go for that. I think it's quite a logical representation - agreed that `lib.rs` should be for the public API only

Reply:
Split performed